packages/cozy-client/src/models/trigger.js

```diff
@@ -59,7 +59,7 @@ const triggerStates = {
   /**
    * Returns the type of the last error to occur
    *
-   * @param {import('../types').TriggersDoctype} triggerState - trigger object
+   * @param {import('../types').IOCozyTrigger} triggerState - trigger object
```

Comment:
Are you sur that is a triggerState here? 

The method reads: 
```
getLastErrorType: triggerState =>
    get(triggerState, 'current_state.last_error')
```

But `current_state` is a property of `TriggersDocument`if it was really a triggerState that was passed, then we'll just returned `  get(triggerState, 'last_error')`

Reply:
OK I got it, TS is lost because of lodash `get`. If you change it by `triggerState?.current_state...` you'll see the error. 

Also, I don't know why, but this `current_state` property is not generated in the `types.d.ts` file. I had to change the order inside the types.js to be able to make it works.
```
/**
 * @typedef {object} TriggersDocument
 * @property {String} [_id] - document identifier
 * @property {String} type - type of the trigger. Can be "at", "cron", "event", "every", "in", "webhook", "client"
 * @property {String} worker - type of worker. Can be "konnector" or "sendmail"
 * @property {object} message - Parameters to pass to the the worker. For example, when the worker is set to konnector, message contains the related konnector and the related account.
 * @property {TriggerState} [current_state] - state of the last executed jobs related to this trigger
 * @property {String} [arguments] - Arguments related to the type attribute. For example it's a cron configuration when the type is set to @cron.
 *
 * @typedef {CozyClientDocument & TriggersDocument} IOCozyTrigger - An io.cozy.konnectors document
 */
```

Reply:
(maybe because of `@cron`)

Reply:
I removed all the get() calls in this file and renamed the triggerState variable everywhere since it is actually a trigger document. I then fixed the corresponding jsdoc issues